cmake/ImportBSON.cmake

```diff
@@ -0,0 +1,170 @@
+   configuration of libmongocrypt. At find_package() time, we can resolve these interface targets
+   to link to the appropriate libbson based on the build configurations of libmongocrypt.
+
+   mongo::mongocrypt must link to _mongocrypt::libbson_for_shared, and mongo::mongocrypt_static must
+   link to _mongocrypt::libbson_for_static.
+
+   At configure+build time, these target will create BUILD_INTERFACE-only usage requirements
+   appropriate for libmongocrypt to build against a libbson. Once these targets are installed,
+   they retain no usage requirements defined here.
+
+   Instead, the installed version of these targets will be manipulated in mongocrypt-config.cmake
+   based on user settings and build configuration options of the installed libmongocrypt in order
+   to ensure that users have satisfied the linking requirements of libmongocrypt.
+   Refer to mongocrypt-config.cmake for more information
+
+   This file calls add_subdirectory(EXCLUDE_FROM_ALL) on a mongo-c-driver project directory. This
+   will expose libbson targets that we can link and use for the libmongocrypt build.
+
+   The boolean option USE_SHARED_LIBBSON controls the behavior of libbson_for_shared:
+
+   If USE_SHARED_LIBBSON=FALSE:
+
+   - libbson_for_shared will transitively link the static libbson from the MONGOCRYPT_MONGOC_DIR.
+   - The result is that mongo::mongocrypt (which is a SHARED library) will have the translation
+     units of libbson directly embedded into the resulting binary.
+   - The symbols from libbson that are merged into mongo::mongocrypt will be supressed using
+     linker scripts such that consumers of mongo::mongocrypt will not see the libbson symbols that
+     were statically linked into the shared library. This allows consumers to link against a
+     completely independent libbson without interfering with the libbson symbols that were merged
+     into mongo::mongocrypt
+   - The installed libbson_for_shared will have no usage requirements.
+
+   If USE_SHARED_LIBBSON=TRUE:
+
+   - libbson_for_shared will transitively use the shared libbson library from
+     the MONGOCRYPT_MONGOC_DIR.
+   - mongo::mongocrypt will be built with a dynamic link requirement on a libbson dynamic
+     library, which must be resolved at runtime by consumers. The translation units from the
+     MONGOCRYPT_MONGOC_DIR *will not* be included in the mongo::mongocrypt library.
+   - The installed libbson_for_shared will dynamically link to a libbson on the user's system by
+     using a find_library() call.
+
+   In both of the above cases, libbson_for_static will require that the final consumer
+   provide their own definitions of the libbson symbols, regardless of the value
+   of USE_SHARED_LIBBSON.
+]]
+
+include(CheckCSourceCompiles)
+include(CMakePushCheckState)
+
+cmake_push_check_state ()
+   # Even though we aren't going to use the system's libbson, try to detect whether it has
```

Comment:
To support the building of Debian packages and RPMs, there has to be an option to build against the system-installed libbson.

Reply:
[See this comment](https://github.com/mongodb/libmongocrypt/pull/365/files#r896084394)